rudder-templates/src/main/scala/com/normation/templates/FillTemplatesService.scala

```diff
@@ -72,8 +72,38 @@ final case class STVariable(
   , isSystem  :Boolean
 )
 
+
 class FillTemplatesService extends Loggable {
 
+  /*
+   * The cache is managed template by template
+   * If a content is not in the cache, it will create the StringTemplate instance, and return it
+   *
+   */
+  private[this] var cache = Map[String,StringTemplate]()
+
+
+  def clearCache(): Unit = this.synchronized {
+    cache = Map()
+  }
+
+  def getTemplateFromContent(templateName: String,content: String): Box[StringTemplate] = this.synchronized {
```

Comment:
i doubled checked - the templateName is indeed the destination of the file, note its source. Which is pretty confusing

Reply:
i dugged into the detail, to see how much ram it took to store the parsed stringtemplate
I have less than 1MB of antlr related class in the heapdump, which is peanuts compared to anything else. Granted, it contains a lot of java instance, but including ALL java instances, I barely reach 200MB (100 MB being Object, 65MB being String)
so i'm pretty convinced the memory usage is not problematic there (at least, we have easier battle to fight to save memory)